Unithereum.CodeGen/Config.cs

```diff
@@ -0,0 +1,403 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using UnityEngine;
+using Debug = UnityEngine.Debug;
+
+namespace Unithereum.CodeGen
+{
+    /// <summary>
+    /// Config class of <see cref="CodeGen" />.
+    /// See README to find an example codegen.config.json file.
+    /// For actual codegen process, refer <see cref="UnithereumCodeGenProcessor"/>.
+    /// </summary>
+    public class Config
+    {
+        /// <summary>
+        /// Absolute path to `dotnet` executable.
+        /// </summary>
+        public string DotnetPath { get; }
+
+        /// <summary>
+        /// Namespace prefix for generated contract service classes.
+        /// <para>Default: (Application.productName).ContractServices</para>
+        /// </summary>
+        public string NamespacePrefix { get; }
+
+        /// <summary>
+        /// Relative path to `Asset` directory for generated contract service class files.
+        /// <para>Default: ContractServices</para>
+        /// </summary>
+        public string OutputDir { get; }
+
+        /// <summary>
+        /// Relative path to Unity project directory where to find `.abi` file inputs.
+        /// <para>Default: Assets</para>
+        /// <para>
+        /// WARNING: Where given path is not inside `Assets/` directory,
+        /// automatic file change detection and generation won't work.
+        /// You may need to manually trigger `Unithereum > Regenerate All...` menu.
+        /// Use with caution.
+        /// </para>
+        /// </summary>
+        public string ContractsDir { get; }
+
+        [JsonConstructor]
+        public Config(
+            [Optional] string? dotnetPath,
+            [Optional] string? namespacePrefix,
+            [Optional] string? outputDir,
+            [Optional] string? contractsDir
+        )
+        {
+            const string defaultName = "ContractServices";
+
+            if (dotnetPath != null && !File.Exists(dotnetPath))
+            {
+                throw new InvalidCodeGenConfigurationException(
+                    message: "`dotnet` executable doesn't exist at given path.",
+                    key: nameof(dotnetPath),
+                    value: dotnetPath
+                );
+            }
+
+            if (namespacePrefix != null && !ValidateNamespacePrefix(namespacePrefix))
+            {
+                throw new InvalidCodeGenConfigurationException(
+                    message: "Use proper C# namespace identifier.",
```

Comment:
```suggestion
                    message: $"{nameof(namespacePrefix)} does not consist of valid C# identifier string.",
```

Might be more helpful for people seeing the message.

Reply:
When `InvalidCodeGenConfigurationException` with `key` argument given, `ToString` will automatically add which key/value caused exception to the message.
Please refer: https://github.com/planetarium/Unithereum/blob/4bb57cb505c707dfd28b561910d7afa7d224f35f/Unithereum.CodeGen/Config.cs#L349-L359

Reply:
Fair enough. 😉